tests/lax_test.py

```diff
@@ -1161,6 +1161,58 @@ def testDotGeneralAgainstNumpy(self, lhs_shape, rhs_shape, dtype,
     numpy_op = lambda x, y: lax_reference.dot_general(x, y, dimension_numbers)
     self._CheckAgainstNumpy(numpy_op, op, args_maker)
 
+  def _testRaggedDot(self, m, k, n, num_groups, dtype, test_cb):
+    """Tests ragged_dot.
+
+    Creates input data and calls test_cb that runs the ragged_dot on the inputs and checks expectations.
+
+    Raises:
+      SkipTest: in the case dtype is not supported.
+    """
+    if np.issubdtype(dtype, np.bool_):
+      # This is due to "reduce_sum does not accept dtype bool".
+      raise SkipTest('ragged_dot for bool not supported')
+    rng = jtu.rand_small(self.rng())
+    lhs_shape = (m, k)
+    rhs_shape = (num_groups, k, n)
+    def group_sizes(m, num_groups):
+      ends_no_final = jnp.sort(np.random.choice(m, size=num_groups - 1))
+      ends = jnp.concatenate([ends_no_final, jnp.array([m], dtype=jnp.int32)])
+      starts = jnp.concatenate([jnp.zeros(1, dtype=jnp.int32), ends_no_final])
+      return ends - starts
+    args_maker = lambda: [rng(lhs_shape, dtype), rng(rhs_shape, dtype), group_sizes(m, num_groups)]
+    test_cb(args_maker)
+
+  @jtu.sample_product(
```

Comment:
I think it would be OK to merge these two combinatorial tests, i.e. do the compile-and-check and the numpy-reference tests in the same test method. We often do both checks in a single test, and merging them generates fewer overall test cases (from which we end up only subsampling).

Reply:
Good idea, thank you; merged the tests into one.